### PR TITLE
Added updateStrategy to DaemonSet mode

### PR DESCRIPTION
**Pull request by yuriolisa**
**Description:** <Describe what has changed.>
Added updateStrategy field regarding the way existing DaemonSet pods will be replaced to new pods.

**Link to tracking Issue:** <Issue number if applicable>
Resolves: #2107 
**Testing:** <Describe what testing was performed and which tests were added.>
Created unit tests and e2e tests to check if these fields have been properly populated.
**Documentation:** <Describe the documentation added.>
https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/daemon-set-v1/#DaemonSetSpec

### CODE REVIEW

**Comment by yuriolisa** [comment]
@frzifus, WDYT to include `updateStrategy` for DaemonSet and Statefulset? So it will be like: 
`statefulsetUpdateStrategy` and `daemonsetUpdateStrategy`, or if you have better suggestions, I'm all ears.

File: apis/v1alpha1/opentelemetrycollector_types.go [hunk 1]
```diff
@@ -275,6 +276,11 @@ type OpenTelemetryCollectorSpec struct {
 	// object, which shall be mounted into the Collector Pods.
 	// Each ConfigMap will be added to the Collector's Deployments as a volume named `configmap-<configmap-name>`.
 	ConfigMaps []ConfigMapsSpec `json:"configmaps,omitempty"`
+	// UpdateStrategy represents the strategy the operator will take replacing existing DaemonSet pods with new pods
+	// https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/daemon-set-v1/#DaemonSetSpec
+	// This is only applicable to Daemonset mode.
```
Comment (frzifus) [inline apis/v1alpha1/opentelemetrycollector_types.go:281]: Does it make sense to validate this in the validation webhook?

**Review by frzifus — COMMENTED** [review]
LGTM, just one tiny thing :)

File: apis/v1alpha1/opentelemetrycollector_types.go (hunk 1 above)
Comment (yuriolisa) [reply apis/v1alpha1/opentelemetrycollector_types.go:281]: Good point! I'll add a validation to enable this field only for DaemonSet mode.

Comment (yuriolisa) [reply apis/v1alpha1/opentelemetrycollector_types.go:281]: Just added the validation you mentioned. Could you please have a look?

**Review by frzifus — APPROVED** [review]

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**